src/main/target/common_post.h

```diff
@@ -26,3 +26,62 @@
 # undef VTX_SMARTAUDIO
 # undef VTX_TRAMP
 #endif
+
+// Forced config of USE_SOFTSERIAL{1,2} in common.h makes SERIAL_PORT_COUNT
+// defined in target.h invalid. Count them and re-define SERIAL_PORT_COUNT.
+#ifdef USE_VCP
+#  define N_VCP 1
+#else
+#  define N_VCP 0
+#endif
+
+#ifdef USE_UART1
+  #define N_UART1 1
+#else
+  #define N_UART1 0
+#endif
+
+#ifdef USE_UART2
+  #define N_UART2 1
+#else
+  #define N_UART2 0
+#endif
+
+#ifdef USE_UART3
+  #define N_UART3 1
+#else
+  #define N_UART3 0
+#endif
+
+#ifdef USE_UART4
+  #define N_UART4 1
+#else
+  #define N_UART4 0
+#endif
+
+#ifdef USE_UART5
+  #define N_UART5 1
+#else
+  #define N_UART5 0
+#endif
+
+#ifdef USE_UART6
+  #define N_UART6 1
+#else
+  #define N_UART6 0
+#endif
+
+#ifdef USE_SOFTSERIAL1
+  #define N_SSERIAL1 1
+#else
+  #define N_SSERIAL1 0
+#endif
+
+#ifdef USE_SOFTSERIAL2
+  #define N_SSERIAL2 1
+#else
+  #define N_SSERIAL2 0
+#endif
+
+#undef SERIAL_PORT_COUNT
```

Comment:
Wouldn't it be easier to maintain if we just removed `SERIAL_PORT_COUNT` from the targets and defined it only here?